File: src/main/java/com/app/umkaSchool/service/TeacherService.java
package com.app.umkaSchool.service;

import com.app.umkaSchool.dto.teacher.CreateTeacherRequest;
import com.app.umkaSchool.dto.teacher.TeacherResponse;
import com.app.umkaSchool.dto.teacher.UpdateTeacherRequest;
import com.app.umkaSchool.model.Teacher;

import java.util.List;
import java.util.UUID;

public interface TeacherService {
    TeacherResponse createTeacher(CreateTeacherRequest request);

    TeacherResponse updateTeacher(UUID teacherId, UpdateTeacherRequest request);

    TeacherResponse getTeacherById(UUID teacherId);

    TeacherResponse getTeacherByUserId(UUID userId);

    List<TeacherResponse> getAllTeachers();

    void deleteTeacher(UUID teacherId);

    Teacher getTeacherEntity(UUID teacherId);
}

File: frontend/src/services/teacher.service.ts
import axiosInstance from './axios.config';
import { Teacher, CreateTeacherPayload, UpdateTeacherPayload } from '../types/teacher';

export const teacherService = {
    getTeacherByUserId: async (userId: string): Promise<Teacher> => {
        const response = await axiosInstance.get<Teacher>(`/api/teachers/user/${userId}`);
        return response.data;
    },

    getTeacherById: async (teacherId: string): Promise<Teacher> => {
        const response = await axiosInstance.get<Teacher>(`/api/teachers/${teacherId}`);
        return response.data;
    },

    getAllTeachers: async (): Promise<Teacher[]> => {
        const response = await axiosInstance.get<Teacher[]>('/api/teachers');
        return response.data;
    },

    createTeacher: async (payload: CreateTeacherPayload): Promise<Teacher> => {
        const response = await axiosInstance.post<Teacher>('/api/teachers', payload);
        return response.data;
    },

    updateTeacher: async (teacherId: string, payload: UpdateTeacherPayload): Promise<Teacher> => {
        const response = await axiosInstance.put<Teacher>(`/api/teachers/${teacherId}`, payload);
        return response.data;
    },

    deleteTeacher: async (teacherId: string): Promise<void> => {
        await axiosInstance.delete(`/api/teachers/${teacherId}`);
    },
};

